node/pkg/watchers/solana/shim.go

```diff
@@ -290,16 +311,21 @@ func (s *SolanaWatcher) shimProcessRest(
 		}
 
 		if verifiedCoreEvent && messageEvent != nil {
-			break
+			if !isTopLevel {
+				// We found what we are looking for, and additional shim events in the inner instruction case are allowed, so we can stop looking.
```

Comment:
Shouldn't the loop continue in case there are multiple `verifiedCoreEvent` occurrences? If not, it would be good to reword this to something like "In the integrator (inner instruction) case, additional core event and shim event pairs are allowed."

Basically my concern with this function is to make sure that:
* there's a one-to-one correspondance of core events and shim events in all cases (i.e. they're always matched)
* there are no "extra events" -- e.g. in the Direct/top-level case, there should only be one core event and one shim event

Reply:
I beefed up the comment.